src/commands/apm_cmds/publish.js

```diff
@@ -392,9 +398,22 @@ exports.task = function ({
       skip: () => onlyContent && !module.path, // TODO: If onlyContent has been set, get previous version's artifact
       task: async (ctx, task) => {
         const dir = onlyArtifacts ? cwd : ctx.pathToPublish
-        const artifact = await generateApplicationArtifact(web3, cwd, dir, module, contract, reporter)
 
-        return `Saved artifact in ${dir}/artifact.json`
+        if (onlyContent && !pathExistsSync(`${dir}/${ARTIFACT_FILE}`)) {
```

Comment:
In a publishing without `--only-content`, which means the contract is being updated, we should always generate a new artifact even if it exists already.

With the current changes, it would never generate artifacts unless the `--only-content` flag is provided.

Reply:
👍